can/interfaces/vector/canlib.py

```diff
@@ -96,18 +102,46 @@ def __init__(self, channel, can_filters=None, poll_interval=0.01,
 
         permission_mask = vxlapi.XLaccess()
         # Set mask to request channel init permission if needed
-        if bitrate:
+        if bitrate or fd:
             permission_mask.value = self.mask
-        vxlapi.xlOpenPort(self.port_handle, self._app_name, self.mask,
-                          permission_mask, rx_queue_size,
-                          vxlapi.XL_INTERFACE_VERSION, vxlapi.XL_BUS_TYPE_CAN)
+        if fd:
+            vxlapi.xlOpenPort(self.port_handle, self._app_name, self.mask,
+                              permission_mask, 16000,
+                              vxlapi.XL_INTERFACE_VERSION_V4, vxlapi.XL_BUS_TYPE_CAN)
+        else:
+            vxlapi.xlOpenPort(self.port_handle, self._app_name, self.mask,
+                              permission_mask, rx_queue_size,
+                              vxlapi.XL_INTERFACE_VERSION, vxlapi.XL_BUS_TYPE_CAN)
         LOG.debug(
             'Open Port: PortHandle: %d, PermissionMask: 0x%X',
             self.port_handle.value, permission_mask.value)
-        if bitrate:
-            if permission_mask.value != self.mask:
-                LOG.info('Can not set bitrate since no init access')
-            vxlapi.xlCanSetChannelBitrate(self.port_handle, permission_mask, bitrate)
+
+        if permission_mask.value == self.mask:
+            if fd:
+                self.canFdConf = vxlapi.XLcanFdConf()
+                if bitrate:
+                    self.canFdConf.arbitrationBitRate = ctypes.c_uint(bitrate)
+                else:
+                    self.canFdConf.arbitrationBitRate = ctypes.c_uint(500000)
+                self.canFdConf.sjwAbr = ctypes.c_uint(2)
```

Comment:
Do these need to be different for different bitrates? Should they be parameters that the user can set?

Reply:
Absolutely!

Reply:
By the way: Do you know whether ctypes.c_uint is necessary?

Reply:
It’s not necessary. The types have already been defined for the struct or the function.